mobile/cmp/popover/Popover.js

```diff
@@ -0,0 +1,196 @@
+import classNames from 'classnames';
+import PT from 'prop-types';
+
+import './Popover.scss';
+
+/**
+ * Popovers display floating content next to a target element.
+ *
+ * The API is based on a stripped-down version of Blueprint's Popover component
+ * that is used on Desktop. Popover is built on top of the Popper.js library.
+ *
+ * @see https://popper.js.org/
+ */
+export const [Popover, popover] = hoistCmp.withFactory({
+    displayName: 'Popover',
+    model: false,
+    className: 'xh-popover',
+
+    render({
+        className,
+        target,
+        content,
+        isOpen,
+        onInteraction,
+        withMask = false,
+        position = 'auto',
+        popperOptions
+    }) {
+        throwIf(!isValidElement(target), 'Popover requires a `target` component');
+        throwIf(!isValidElement(content), 'Popover requires a `content` component');
+
+        const impl = useLocalModel(() => new LocalModel()),
+            popper = usePopper(impl.targetEl, impl.contentEl, {
+                placement: impl.menuPositionToPlacement(position),
+                strategy: 'fixed',
+                modifiers: [
+                    {name: 'preventOverflow', options: {padding: 10}}
+                ],
+                ...popperOptions
+            });
+
+        // Respect controlled mode prop
+        if (!isNil(isOpen)) {
+            impl.setControlledMode(isOpen, onInteraction);
+        }
+
+        return div({
+            className,
+            items: [
+                div({
+                    ref: impl.targetRef,
+                    className: 'xh-popover__target-wrapper',
+                    item: target,
+                    onClick: () => impl.toggleOpen()
+                }),
+                div({
+                    ref: impl.contentRef,
+                    omit: !impl.isOpen,
+                    style: popper?.styles?.popper,
+                    className: 'xh-popover__content-wrapper',
+                    items: content
+                }),
+                div({
+                    omit: !impl.isOpen,
+                    className: classNames(
+                        'xh-popover__content-overlay',
+                        withMask ? 'xh-popover__content-overlay--mask' : null
+                    ),
+                    onClick: () => impl.setIsOpen(false)
+                })
+            ]
+        });
+    }
+});
+
+Popover.propTypes = {
+    /** Component to display inside the popover */
+    content: PT.element,
+
+    /** The target to which the popover content is attached */
+    target: PT.element,
+
+    /** Whether the popover is visible. Passing this prop puts the popover in controlled mode */
+    isOpen: PT.bool,
+
+    /**
+     * Callback invoked in controlled mode when the popover open state _would_ change due to user interaction.
+     * Receives (nextOpenState: boolean)
+     */
+    onInteraction: PT.func,
+
+    /** Whether to display a mask behind the popover */
+    withMask: PT.bool,
```

Comment:
would it be useful to call the property `mask` and have users be able to specify the mask itself, if they wanted to (or a boolean) we do this elsewheere in the API.

Reply:
I think being able to specify a mask itself will confuse things... you don't want to show a spinner or tie this to a load model, etc.

The overlay is always there, it's necessary to be able to dismiss the popover. This prop is really just whether or not to color the overlay like a mask.

Reply:
Blueprint uses `hasBackdrop` for what sounds like the analogous setting on their `Drawer` component - that might be a good option here, to distinguish from our other usages of mask.  (I would probably go with `withBackdrop` or simply `backdrop`.)

I don't feel strongly about it - the prop as its named now avoids introducing a new term and is pretty clear - I don't think it's that likely that anyone would interpret it as putting a mask _over_ the popover, although maybe...

Reply:
I like renaming it to `backdrop` to avoid any potential ambiguity - will make that change now